### PR TITLE
Update demo to latest +TCP dev/IPv6_integration

### PR DESCRIPTION
**Pull request by tony-josi-aws**
Description
-----------
This PR updates the +TCP demos to be compatible with the latest changes related to picking latest cmake changes from +TCP mainline branch to +TCP dev/IPv6_integration.

Test Steps
-----------
Build and run the demos on Jenkins.

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [ ] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: FreeRTOS-Plus/Demo/Common/Demo_IP_Protocols/Common/FreeRTOS_TCP_server.c [hunk 1]
```diff
@@ -91,7 +91,16 @@ SocketSet_t xSocketSet;
 
 					if( xSocket != FREERTOS_INVALID_SOCKET )
 					{
-						xAddress.sin_addr = FreeRTOS_GetIPAddress(); // Single NIC, currently not used
+						#if defined( ipconfigIPv4_BACKWARD_COMPATIBLE ) && ( ipconfigIPv4_BACKWARD_COMPATIBLE == 0 )
+						{
+							xAddress.sin_address.ulIP_IPv4 = FreeRTOS_GetIPAddress(); // Single NIC, currently not used
+						}
```
Comment (shubnil) [inline FreeRTOS-Plus/Demo/Common/Demo_IP_Protocols/Common/FreeRTOS_TCP_server.c:97]: Please change the comment to /* */ format.

File: FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_Minimal_Windows_Simulator/DemoTasks/SimpleUDPClientAndServer.c [hunk 2]
```diff
@@ -329,10 +344,12 @@ Socket_t xListeningSocket;
 	after the network is up, so the IP address is valid here. */
 #if defined( ipconfigIPv4_BACKWARD_COMPATIBLE ) && ( ipconfigIPv4_BACKWARD_COMPATIBLE == 0 )
 	FreeRTOS_GetEndPointConfiguration( &ulIPAddress, NULL, NULL, NULL, pxNetworkEndPoints );
+	xBindAddress.sin_address.ulIP_IPv4 = ulIPAddress;
 #else
 	FreeRTOS_GetAddressConfiguration( &ulIPAddress, NULL, NULL, NULL );
-#endif
 	xBindAddress.sin_addr = ulIPAddress;
+#endif
```
Comment (shubnil) [inline FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_Minimal_Windows_Simulator/DemoTasks/SimpleUDPClientAndServer.c:351]: Let us follow the same the format for #endif in all the places. Some places have /* defined( ipconfigIPv4_BACKWARD_COMPATIBLE ) && ( ipconfigIPv4_BACKWARD_COMPATIBLE == 0 ) */ and some do not.

**Review by shubnil — CHANGES_REQUESTED** [review]

File: FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_Echo_Posix/Makefile [hunk 3]
```diff
@@ -52,31 +52,10 @@ SOURCE_FILES += ${FREERTOS_DIR}/Source/portable/MemMang/heap_3.c
 
 
 # FreeRTOS TCP
-SOURCE_FILES += ${FREERTOS_PLUS_DIR}/Source/FreeRTOS-Plus-TCP/source/FreeRTOS_ARP.c
```
Comment (moninom1) [inline FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_Echo_Posix/Makefile:55]: These makefile changes are added for only 2 demos? Any reason?

Comment (tony-josi-aws) [reply FreeRTOS-Plus/Demo/FreeRTOS_Plus_TCP_Echo_Posix/Makefile:55]: I believe all the other +TCP demos are not based on Makefiles

**Review by shubnil — APPROVED** [review]

**Pull request merged**